### ISSUE TITLE
0 mAP for most classes after exporting with int8=True

### ISSUE THREAD

**Issue by leeeunhyeok1** [ultralytics/ultralytics#22644]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

Hi im using yolov10n for object detection. 

After training end, I exported best.pt to tflite with int8=True with following command: 
!yolo export model=runs/detect/train/weights/best.pt format=tflite device=0 int8=True  fraction=1.0 data=data.yaml

While best_int8.tflite shows similar performance with best.pt, both best_integer_quant.tflite and best_full_integer_quant.tflite gives 0 mAP for all classes except one.

best_integer_quant.tflite and best_full_integer_quant.tflite validation result:
<img width="640" height="674" alt="Image" src="https://github.com/user-attachments/assets/25c1614a-bd8a-46d8-b90e-61ab19cb5408" />

I did the same process with yolov8n, and both models worked fine. 

How can I fix this?  

### Environment

Creating new Ultralytics Settings v0.0.6 file ✅ 
View Ultralytics Settings with 'yolo settings' or at '/root/.config/Ultralytics/settings.json'
Update Settings with 'yolo settings key=value', i.e. 'yolo settings runs_dir=path/to/dir'. For help see https://docs.ultralytics.com/quickstart/#ultralytics-settings.
Ultralytics 8.3.227 🚀 Python-3.12.12 torch-2.8.0+cu126 CUDA:0 (Tesla T4, 15095MiB)
Setup complete ✅ (2 CPUs, 12.7 GB RAM, 39.4/112.6 GB disk)

OS                     Linux-6.6.105+-x86_64-with-glibc2.35
Environment            Colab
Python                 3.12.12
Install                pip
Path                   /usr/local/lib/python3.12/dist-packages/ultralytics
RAM                    12.67 GB
Disk                   39.4/112.6 GB
CPU                    Intel Xeon CPU @ 2.00GHz
CPU count              2
GPU                    Tesla T4, 15095MiB
GPU count              1
CUDA                   12.6

numpy                  ✅ 2.0.2>=1.23.0
matplotlib             ✅ 3.10.0>=3.3.0
opencv-python          ✅ 4.12.0.88>=4.6.0
pillow                 ✅ 11.3.0>=7.1.2
pyyaml                 ✅ 6.0.3>=5.3.1
requests               ✅ 2.32.4>=2.23.0
scipy                  ✅ 1.16.3>=1.4.1
torch                  ✅ 2.8.0+cu126>=1.8.0
torch                  ✅ 2.8.0+cu126!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision            ✅ 0.23.0+cu126>=0.9.0
psutil                 ✅ 5.9.5
polars                 ✅ 1.25.2
ultralytics-thop       ✅ 2.0.18>=2.0.18

### Minimal Reproducible Example

my training command:
```
!yolo task=detect mode=train model="yolov10n.yaml" data="data.yaml" \
epochs=1500 imgsz=640 batch=16 device=0 optimizer="AdamW" seed=1337 augment=False \
fliplr=0.0 flipud=0.0 mosaic=0.0 translate=0.1 scale=0.0 erasing=0.0 hsv_h=0.0 \
 hsv_s=0.0 hsv_v=0.0 auto_augment=None copy_paste=0.0 close_mosaic=0
```

export command:
```
!yolo export model=runs/detect/train/weights/best.pt format=tflite device=0 int8=True  fraction=1.0 data=data.yaml
```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @leeeunhyeok1, thanks for opening this issue and for the detailed context 🚀 This is an automated triage message—an Ultralytics engineer will review and assist you shortly.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it. It looks like you’ve shared good repro steps—thank you! To help us pinpoint the INT8 TFLite mAP discrepancy further, please also include:
-Exact export logs (console output) for the int8 export, including any warnings
-Validation command and metrics for best.pt and each TFLite artifact you tested
-A few sample images where the issue occurs and the expected detections
-Details on calibration data used during INT8 export (dataset size, representativeness)
-The TFLite runtime environment and interpreter flags you used for evaluation

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [Python>=3.8](https://www.python.org/) environment with [PyTorch>=1.8](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thanks again for the report 🙏 An engineer will follow up soon to help investigate the INT8 export behavior you observed.

**Comment by Y-T-G** [comment]
Static quantization probably doesn't work for end2end models like YOLOv10.
